impeller/renderer/backend/gles/blit_command_gles.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/gles/blit_command_gles.h"
+
+#include "flutter/fml/closure.h"
+#include "impeller/base/validation.h"
+#include "impeller/renderer/backend/gles/texture_gles.h"
+
+namespace impeller {
+
+BlitEncodeGLES::~BlitEncodeGLES() = default;
+
+static void DeleteFBO(const ProcTableGLES& gl, GLuint fbo, GLenum type) {
+  if (fbo != GL_NONE) {
+    gl.BindFramebuffer(type, GL_NONE);
+    gl.DeleteFramebuffers(1u, &fbo);
+  }
+};
+
+static std::optional<GLuint> ConfigureFBO(
+    const ProcTableGLES& gl,
+    const std::shared_ptr<Texture>& texture,
+    GLenum fbo_type) {
+  auto handle = TextureGLES::Cast(texture.get())->GetGLHandle();
+  if (!handle.has_value()) {
+    return std::nullopt;
+  }
+
+  if (TextureGLES::Cast(*texture).IsWrapped()) {
+    // The texture is attached to the default FBO, so there's no need to
+    // create/configure one.
+    gl.BindFramebuffer(fbo_type, 0);
+    return 0;
+  }
+
+  GLuint fbo;
+  gl.GenFramebuffers(1u, &fbo);
+  gl.BindFramebuffer(fbo_type, fbo);
+
+  if (!TextureGLES::Cast(*texture).SetAsFramebufferAttachment(
+          fbo_type, fbo, TextureGLES::AttachmentPoint::kColor0)) {
+    VALIDATION_LOG << "Could not attach texture to framebuffer.";
+    DeleteFBO(gl, fbo, fbo_type);
+    return std::nullopt;
+  }
+
+  if (gl.CheckFramebufferStatus(fbo_type) != GL_FRAMEBUFFER_COMPLETE) {
+    VALIDATION_LOG << "Could not create a complete frambuffer.";
+    DeleteFBO(gl, fbo, fbo_type);
+    return std::nullopt;
+  }
+
+  return fbo;
+};
+
+BlitCopyTextureToTextureCommandGLES::~BlitCopyTextureToTextureCommandGLES() =
+    default;
+
+std::string BlitCopyTextureToTextureCommandGLES::GetLabel() const {
+  return label;
+}
+
+bool BlitCopyTextureToTextureCommandGLES::Encode(
+    const ReactorGLES& reactor) const {
+  const auto& gl = reactor.GetProcTable();
+
+  // glBlitFramebuffer is a GLES3 proc. Since we target GLES2, we need to
+  // emulate the blit when it's not available in the driver.
+  if (!gl.BlitFramebuffer.IsAvailable()) {
+    // TODO(bdero): Emulate the blit using a raster draw call here.
+    FML_LOG(ERROR) << "Texture blit fallback not implemented yet for GLES2.";
+    return true;
```

Comment:
Return `false`?

Reply:
Eh, I suppose we can't really use CopyTextureToTexture for real until we have this fallback in place anyhow, so might as well fail loudly by failing the whole frame. Done.